cmd/root.go

```diff
@@ -2,21 +2,41 @@ package cmd
 
 import (
 	"github.com/spf13/cobra"
+	"kool-dev/kool/environment"
+	"os"
 )
 
 // CobraRunFN Cobra command run function
 type CobraRunFN func(*cobra.Command, []string)
 
 var version string = "0.0.0-dev"
 
-var rootCmd = &cobra.Command{
-	Use:   "kool",
-	Short: "kool - Kool stuff",
-	Long: `An easy and robust software development environment
+var rootCmd = NewRootCmd(environment.NewEnvStorage())
+
+// NewRootCmd creates the root command
+func NewRootCmd(envStorage environment.EnvStorage) (cmd *cobra.Command) {
+	cmd = &cobra.Command{
+		Use:   "kool",
+		Short: "kool - Kool stuff",
+		Long: `An easy and robust software development environment
 tool helping you from project creation until deployment.
 Complete documentation is available at https://kool.dev/docs`,
-	Version:           version,
-	DisableAutoGenTag: true,
+		Version:           version,
+		DisableAutoGenTag: true,
+		PersistentPreRun: func(cmf *cobra.Command, args []string) {
+			if verbose := cmf.Flags().Lookup("verbose"); verbose != nil && verbose.Value.String() == "true" {
+				envStorage.Set("KOOL_VERBOSE", verbose.Value.String())
+			}
+
+			if asuser := cmf.Flags().Lookup("asuser"); asuser != nil && asuser.Value.String() != "0" {
+				envStorage.Set("KOOL_ASUSER", asuser.Value.String())
+			}
+		},
+	}
+
+	cmd.PersistentFlags().Bool("verbose", false, "execute commands with verbosity")
+	cmd.PersistentFlags().Int("asuser", os.Getuid(), "execute commands as user")
```

Comment:
The `--verbose` flag I guess is worth having it persisted, but the `asuser` do not make sense for most of the commands we have, and also it is under review on #166. So I would say we should not touch the `asuser` at this moment, removing it from this change set that otherwise looks good.